src/main/java/net/neoforged/bus/EventBus.java

```diff
@@ -311,14 +312,14 @@ public void unregister(Object object)
     }
 
     @Override
-    public boolean post(Event event) {
+    public <T extends Event> T post(T event) {
         return post(event, (IEventListener::invoke));
     }
 
     @Override
-    public boolean post(Event event, IEventBusInvokeDispatcher wrapper)
+    public < T extends Event> T post(T event, IEventBusInvokeDispatcher wrapper)
     {
-        if (shutdown) return false;
+        if (shutdown) return event;
```

Comment:
Should we error if an event is posted to a shutdown event bus?

Reply:
I don't know. It seems like a good idea! Would there be a drawback to this? I suppose we could try to throw an exception

Reply:
I'd argue it prevents issues hard to debug when the bus was stopped mid-runtime, but events are still being fired.

Reply:
When does the bus get stopped? Do mods have to account for that to prevent shutdown crashes? That is my only concern

Reply:
Currently never if IDEA isn't lying to me, but there is a timeframe during startup where the Forge bus is not yet started, which would be made very obvious by this measure

Reply:
It would be nice to throw an exception when the bus wasn't started yet, IMO. Would probably prevent a lot of mistakes.

Reply:
> When does the bus get stopped?
The bus can also be stopped by anyone quite invisibly (just one logging line in a log that's otherwise a few megabytes).

Reply:
How about we remove the `stop()` method entirely? Forge doesn't use it.

Reply:
Did that.